### PR TITLE
feat: change chainId to u64

### PR DESCRIPTION
**Pull request by AnastasiiaVashchuk**

### CODE REVIEW

File: core/lib/basic_types/src/lib.rs [hunk 1]
```diff
@@ -76,6 +76,92 @@ impl TryFrom<U256> for AccountTreeId {
...
+    }
+}
+
+impl FromStr for L2ChainId {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        // Parse the string as a U64
+        // try to parse as decimal first
+        let u64 = match U64::from_dec_str(s) {
```
Comment (Deniallugo) [inline core/lib/basic_types/src/lib.rs:116]: it's very confusing name

File: core/lib/config/src/configs/chain.rs [hunk 2]
```diff
@@ -202,7 +202,7 @@ mod tests {
             network: NetworkConfig {
                 network: "localhost".parse().unwrap(),
                 zksync_network: "localhost".to_string(),
-                zksync_network_id: 270,
+                zksync_network_id: zksync_basic_types::L2ChainId(270),
```
Comment (Deniallugo) [inline core/lib/config/src/configs/chain.rs:205]: you don't need to specify `zksync_basic_types`

File: core/lib/basic_types/src/lib.rs [hunk 3]
```diff
@@ -78,32 +78,15 @@ impl TryFrom<U256> for AccountTreeId {
...
-        };
-
-        if u64.as_u64() > L2ChainId::max().0 {
-            return Err(de::Error::custom(format!(
-                "Too big chain ID. MAX: {}",
-                L2ChainId::max().0
-            )));
-        }
-        Ok(L2ChainId(u64.as_u64()))
+        s.parse().map_err(de::Error::custom)
```
Comment (Deniallugo) [inline core/lib/basic_types/src/lib.rs:89]: why are you not checking the max value anymore?

File: core/lib/basic_types/src/lib.rs [hunk 4]
```diff
@@ -162,6 +149,12 @@ impl TryFrom<u64> for L2ChainId {
     }
 }
 
+impl From<u32> for L2ChainId {
```
Comment (Deniallugo) [inline core/lib/basic_types/src/lib.rs:152]: Why only u32? I think we have to have TryFrom<u64> as well

Comment (AnastasiiaVashchuk) [reply core/lib/basic_types/src/lib.rs:152]: why?
2^32 is < then MAX

File: core/lib/basic_types/src/lib.rs (hunk 3 above)
Comment (AnastasiiaVashchuk) [reply core/lib/basic_types/src/lib.rs:89]: MAX is checked in FromStr, i.e parse() is invoked

File: core/lib/basic_types/src/lib.rs (hunk 4 above)
Comment (Deniallugo) [reply core/lib/basic_types/src/lib.rs:152]: GitHub ate the `u64`

File: core/lib/basic_types/src/lib.rs (hunk 3 above)
Comment (Deniallugo) [reply core/lib/basic_types/src/lib.rs:89]: Let's have a few unittests.

File: core/lib/basic_types/Cargo.toml [hunk 5]
```diff
@@ -12,3 +12,5 @@ categories = ["cryptography"]
 [dependencies]
 web3 = { version= "0.19.0", default-features = false, features = ["http-rustls-tls", "test", "signing"] }
 serde = { version = "1.0", features = ["derive"] }
+serde_json = "1.0.107"
```
Comment (Deniallugo) [inline core/lib/basic_types/Cargo.toml:15]: keep the version just 1.0, otherwise it will be impossible to simple increase the minor patch

File: core/lib/basic_types/src/lib.rs [hunk 6]
```diff
@@ -139,15 +212,78 @@ impl Default for L1BlockNumber {
     }
 }
 
-impl Default for L2ChainId {
-    fn default() -> Self {
-        Self(270)
```
Comment (Deniallugo) [inline core/lib/basic_types/src/lib.rs:144]: Why did you remove the default?

Comment (AnastasiiaVashchuk) [reply core/lib/basic_types/src/lib.rs:144]: take a look on like 131

Comment (Deniallugo) [reply core/lib/basic_types/src/lib.rs:144]:
```
#[allow(clippy::derivable_impls)]
impl Default for L1BatchNumber {
    fn default() -> Self {
        Self(0)
    }
}

``` 

What?

Comment (AnastasiiaVashchuk) [reply core/lib/basic_types/src/lib.rs:144]: image

**Review by Deniallugo — APPROVED** [review]
